extensions/kubeapi/projects/list.go

```diff
@@ -0,0 +1,36 @@
+package projects
+
+import (
+	"context"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/pkg/api/scheme"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ListProjects is a helper function that uses the dynamic client to list projects in a specific cluster.
+func ListProjects(client *rancher.Client, namespace string, listOpt metav1.ListOptions) (*v3.ProjectList, error) {
```

Comment:
The same comment that I made: https://github.com/rancher/shepherd/pull/16#discussion_r1443094331

Reply:
As Anu mentioned in the comment ^^ we can address this outside the scope of this PR by creating a qa task.

Reply:
Agreed